// pdf-generator-api-client/dist/model/DataBatchInner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DataBatchInner model module.
 * @module model/DataBatchInner
 * @version 4.0.3
 */
var DataBatchInner = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DataBatchInner</code>.
   * @alias module:model/DataBatchInner
   */
  function DataBatchInner() {
    _classCallCheck(this, DataBatchInner);

    DataBatchInner.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DataBatchInner, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DataBatchInner</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DataBatchInner} obj Optional instance to populate.
     * @return {module:model/DataBatchInner} The populated <code>DataBatchInner</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DataBatchInner();

        if (data.hasOwnProperty('template')) {
          obj['template'] = _ApiClient["default"].convertToType(data['template'], 'Number');
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], Object);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>DataBatchInner</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DataBatchInner</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);

  return DataBatchInner;
}();
/**
 * Template id
 * @member {Number} template
 */


DataBatchInner.prototype['template'] = undefined;
/**
 * JSON data used to replace data fields in the template
 * @member {Object} data
 */

DataBatchInner.prototype['data'] = undefined;
var _default = DataBatchInner;
exports["default"] = _default;
