// pdf-generator-api-client/dist/model/Document.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Document model module.
 * @module model/Document
 * @version 4.0.3
 */
var Document = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Document</code>.
   * @alias module:model/Document
   */
  function Document() {
    _classCallCheck(this, Document);

    Document.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Document, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Document</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Document} obj Optional instance to populate.
     * @return {module:model/Document} The populated <code>Document</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Document();

        if (data.hasOwnProperty('public_id')) {
          obj['public_id'] = _ApiClient["default"].convertToType(data['public_id'], 'String');
        }

        if (data.hasOwnProperty('public_url')) {
          obj['public_url'] = _ApiClient["default"].convertToType(data['public_url'], 'String');
        }

        if (data.hasOwnProperty('template_id')) {
          obj['template_id'] = _ApiClient["default"].convertToType(data['template_id'], 'Number');
        }

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'String');
        }

        if (data.hasOwnProperty('expires_at')) {
          obj['expires_at'] = _ApiClient["default"].convertToType(data['expires_at'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>Document</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Document</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['public_id'] && !(typeof data['public_id'] === 'string' || data['public_id'] instanceof String)) {
        throw new Error("Expected the field `public_id` to be a primitive type in the JSON string but got " + data['public_id']);
      } // ensure the json data is a string


      if (data['public_url'] && !(typeof data['public_url'] === 'string' || data['public_url'] instanceof String)) {
        throw new Error("Expected the field `public_url` to be a primitive type in the JSON string but got " + data['public_url']);
      } // ensure the json data is a string


      if (data['created_at'] && !(typeof data['created_at'] === 'string' || data['created_at'] instanceof String)) {
        throw new Error("Expected the field `created_at` to be a primitive type in the JSON string but got " + data['created_at']);
      } // ensure the json data is a string


      if (data['expires_at'] && !(typeof data['expires_at'] === 'string' || data['expires_at'] instanceof String)) {
        throw new Error("Expected the field `expires_at` to be a primitive type in the JSON string but got " + data['expires_at']);
      }

      return true;
    }
  }]);

  return Document;
}();
/**
 * Document unique identifier
 * @member {String} public_id
 */


Document.prototype['public_id'] = undefined;
/**
 * Public URL of the document
 * @member {String} public_url
 */

Document.prototype['public_url'] = undefined;
/**
 * Template used to generate the document
 * @member {Number} template_id
 */

Document.prototype['template_id'] = undefined;
/**
 * Date time when the document was generated
 * @member {String} created_at
 */

Document.prototype['created_at'] = undefined;
/**
 * Date time when the document url will expire. Document is stored for 30 days.
 * @member {String} expires_at
 */

Document.prototype['expires_at'] = undefined;
var _default = Document;
exports["default"] = _default;
