// pdf-generator-api-client/dist/model/GenerateDocument201Response.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GenerateDocument201ResponseMeta = _interopRequireDefault(require("./GenerateDocument201ResponseMeta"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GenerateDocument201Response model module.
 * @module model/GenerateDocument201Response
 * @version 4.0.3
 */
var GenerateDocument201Response = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GenerateDocument201Response</code>.
   * @alias module:model/GenerateDocument201Response
   */
  function GenerateDocument201Response() {
    _classCallCheck(this, GenerateDocument201Response);

    GenerateDocument201Response.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GenerateDocument201Response, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GenerateDocument201Response</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GenerateDocument201Response} obj Optional instance to populate.
     * @return {module:model/GenerateDocument201Response} The populated <code>GenerateDocument201Response</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GenerateDocument201Response();

        if (data.hasOwnProperty('response')) {
          obj['response'] = _ApiClient["default"].convertToType(data['response'], 'String');
        }

        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _GenerateDocument201ResponseMeta["default"].constructFromObject(data['meta']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GenerateDocument201Response</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GenerateDocument201Response</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['response'] && !(typeof data['response'] === 'string' || data['response'] instanceof String)) {
        throw new Error("Expected the field `response` to be a primitive type in the JSON string but got " + data['response']);
      } // validate the optional field `meta`


      if (data['meta']) {
        // data not null
        _GenerateDocument201ResponseMeta["default"].validateJSON(data['meta']);
      }

      return true;
    }
  }]);

  return GenerateDocument201Response;
}();
/**
 * Base64 encoded document if the output=base64 is used or URL to the document when the output=url is used.
 * @member {String} response
 */


GenerateDocument201Response.prototype['response'] = undefined;
/**
 * @member {module:model/GenerateDocument201ResponseMeta} meta
 */

GenerateDocument201Response.prototype['meta'] = undefined;
var _default = GenerateDocument201Response;
exports["default"] = _default;
