// pdf-generator-api-client/dist/model/GenerateDocumentAsynchronousRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AsyncOutputParam = _interopRequireDefault(require("./AsyncOutputParam"));

var _CallbackParam = _interopRequireDefault(require("./CallbackParam"));

var _FormatParam = _interopRequireDefault(require("./FormatParam"));

var _TemplateParam = _interopRequireDefault(require("./TemplateParam"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GenerateDocumentAsynchronousRequest model module.
 * @module model/GenerateDocumentAsynchronousRequest
 * @version 4.0.3
 */
var GenerateDocumentAsynchronousRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GenerateDocumentAsynchronousRequest</code>.
   * @alias module:model/GenerateDocumentAsynchronousRequest
   */
  function GenerateDocumentAsynchronousRequest() {
    _classCallCheck(this, GenerateDocumentAsynchronousRequest);

    GenerateDocumentAsynchronousRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GenerateDocumentAsynchronousRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GenerateDocumentAsynchronousRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GenerateDocumentAsynchronousRequest} obj Optional instance to populate.
     * @return {module:model/GenerateDocumentAsynchronousRequest} The populated <code>GenerateDocumentAsynchronousRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GenerateDocumentAsynchronousRequest();

        if (data.hasOwnProperty('template')) {
          obj['template'] = _TemplateParam["default"].constructFromObject(data['template']);
        }

        if (data.hasOwnProperty('callback')) {
          obj['callback'] = _CallbackParam["default"].constructFromObject(data['callback']);
        }

        if (data.hasOwnProperty('format')) {
          obj['format'] = _FormatParam["default"].constructFromObject(data['format']);
        }

        if (data.hasOwnProperty('output')) {
          obj['output'] = _AsyncOutputParam["default"].constructFromObject(data['output']);
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GenerateDocumentAsynchronousRequest</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GenerateDocumentAsynchronousRequest</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `template`
      if (data['template']) {
        // data not null
        _TemplateParam["default"].validateJSON(data['template']);
      } // validate the optional field `callback`


      if (data['callback']) {
        // data not null
        _CallbackParam["default"].validateJSON(data['callback']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }

      return true;
    }
  }]);

  return GenerateDocumentAsynchronousRequest;
}();
/**
 * @member {module:model/TemplateParam} template
 */


GenerateDocumentAsynchronousRequest.prototype['template'] = undefined;
/**
 * @member {module:model/CallbackParam} callback
 */

GenerateDocumentAsynchronousRequest.prototype['callback'] = undefined;
/**
 * @member {module:model/FormatParam} format
 */

GenerateDocumentAsynchronousRequest.prototype['format'] = undefined;
/**
 * @member {module:model/AsyncOutputParam} output
 */

GenerateDocumentAsynchronousRequest.prototype['output'] = undefined;
/**
 * Generated document name (optional)
 * @member {String} name
 * @default ''
 */

GenerateDocumentAsynchronousRequest.prototype['name'] = '';
var _default = GenerateDocumentAsynchronousRequest;
exports["default"] = _default;
