// pdf-generator-api-client/dist/model/GenerateDocumentBatchAsynchronousRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AsyncOutputParam = _interopRequireDefault(require("./AsyncOutputParam"));

var _CallbackParam = _interopRequireDefault(require("./CallbackParam"));

var _FormatParam = _interopRequireDefault(require("./FormatParam"));

var _TemplateParam = _interopRequireDefault(require("./TemplateParam"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GenerateDocumentBatchAsynchronousRequest model module.
 * @module model/GenerateDocumentBatchAsynchronousRequest
 * @version 4.0.3
 */
var GenerateDocumentBatchAsynchronousRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GenerateDocumentBatchAsynchronousRequest</code>.
   * @alias module:model/GenerateDocumentBatchAsynchronousRequest
   */
  function GenerateDocumentBatchAsynchronousRequest() {
    _classCallCheck(this, GenerateDocumentBatchAsynchronousRequest);

    GenerateDocumentBatchAsynchronousRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GenerateDocumentBatchAsynchronousRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GenerateDocumentBatchAsynchronousRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GenerateDocumentBatchAsynchronousRequest} obj Optional instance to populate.
     * @return {module:model/GenerateDocumentBatchAsynchronousRequest} The populated <code>GenerateDocumentBatchAsynchronousRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GenerateDocumentBatchAsynchronousRequest();

        if (data.hasOwnProperty('template')) {
          obj['template'] = _ApiClient["default"].convertToType(data['template'], [_TemplateParam["default"]]);
        }

        if (data.hasOwnProperty('callback')) {
          obj['callback'] = _CallbackParam["default"].constructFromObject(data['callback']);
        }

        if (data.hasOwnProperty('format')) {
          obj['format'] = _FormatParam["default"].constructFromObject(data['format']);
        }

        if (data.hasOwnProperty('output')) {
          obj['output'] = _AsyncOutputParam["default"].constructFromObject(data['output']);
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GenerateDocumentBatchAsynchronousRequest</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GenerateDocumentBatchAsynchronousRequest</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['template']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['template'])) {
          throw new Error("Expected the field `template` to be an array in the JSON data but got " + data['template']);
        } // validate the optional field `template` (array)


        var _iterator = _createForOfIteratorHelper(data['template']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _TemplateParam["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // validate the optional field `callback`


      if (data['callback']) {
        // data not null
        _CallbackParam["default"].validateJSON(data['callback']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }

      return true;
    }
  }]);

  return GenerateDocumentBatchAsynchronousRequest;
}();
/**
 * @member {Array.<module:model/TemplateParam>} template
 */


GenerateDocumentBatchAsynchronousRequest.prototype['template'] = undefined;
/**
 * @member {module:model/CallbackParam} callback
 */

GenerateDocumentBatchAsynchronousRequest.prototype['callback'] = undefined;
/**
 * @member {module:model/FormatParam} format
 */

GenerateDocumentBatchAsynchronousRequest.prototype['format'] = undefined;
/**
 * @member {module:model/AsyncOutputParam} output
 */

GenerateDocumentBatchAsynchronousRequest.prototype['output'] = undefined;
/**
 * Generated document name (optional)
 * @member {String} name
 * @default ''
 */

GenerateDocumentBatchAsynchronousRequest.prototype['name'] = '';
var _default = GenerateDocumentBatchAsynchronousRequest;
exports["default"] = _default;
