// pdf-generator-api-client/dist/model/GetTemplates401Response.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetTemplates401Response model module.
 * @module model/GetTemplates401Response
 * @version 4.0.3
 */
var GetTemplates401Response = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetTemplates401Response</code>.
   * @alias module:model/GetTemplates401Response
   */
  function GetTemplates401Response() {
    _classCallCheck(this, GetTemplates401Response);

    GetTemplates401Response.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetTemplates401Response, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetTemplates401Response</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetTemplates401Response} obj Optional instance to populate.
     * @return {module:model/GetTemplates401Response} The populated <code>GetTemplates401Response</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetTemplates401Response();

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetTemplates401Response</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetTemplates401Response</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['message'] && !(typeof data['message'] === 'string' || data['message'] instanceof String)) {
        throw new Error("Expected the field `message` to be a primitive type in the JSON string but got " + data['message']);
      }

      return true;
    }
  }]);

  return GetTemplates401Response;
}();
/**
 * Error description
 * @member {module:model/GetTemplates401Response.MessageEnum} message
 */


GetTemplates401Response.prototype['message'] = undefined;
/**
 * Allowed values for the <code>message</code> property.
 * @enum {String}
 * @readonly
 */

GetTemplates401Response['MessageEnum'] = {
  /**
   * value: "Authentication failed: request expired"
   * @const
   */
  "failed: request expired": "Authentication failed: request expired",

  /**
   * value: "Authentication failed: signature or secret missing"
   * @const
   */
  "failed: signature or secret missing": "Authentication failed: signature or secret missing",

  /**
   * value: "Authentication failed: workspace missing"
   * @const
   */
  "failed: workspace missing": "Authentication failed: workspace missing",

  /**
   * value: "Authentication failed: key missing"
   * @const
   */
  "failed: key missing": "Authentication failed: key missing",

  /**
   * value: "Authentication failed: property iss (issuer) missing in JWT"
   * @const
   */
  "failed: property iss (issuer) missing in JWT": "Authentication failed: property iss (issuer) missing in JWT",

  /**
   * value: "Authentication failed: property sub (subject) missing in JWT"
   * @const
   */
  "failed: property sub (subject) missing in JWT": "Authentication failed: property sub (subject) missing in JWT",

  /**
   * value: "Authentication failed: property exp (expiration time) missing in JWT"
   * @const
   */
  "failed: property exp (expiration time) missing in JWT": "Authentication failed: property exp (expiration time) missing in JWT",

  /**
   * value: "Authentication failed: invalid iss (issuer)"
   * @const
   */
  "failed: invalid iss (issuer)": "Authentication failed: invalid iss (issuer)",

  /**
   * value: "Authentication failed: incorrect signature"
   * @const
   */
  "failed: incorrect signature": "Authentication failed: incorrect signature",

  /**
   * value: "Authentication failed"
   * @const
   */
  "failed": "Authentication failed"
};
var _default = GetTemplates401Response;
exports["default"] = _default;
