// pdf-generator-api-client/dist/model/GetTemplates402Response.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetTemplates402Response model module.
 * @module model/GetTemplates402Response
 * @version 4.0.3
 */
var GetTemplates402Response = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetTemplates402Response</code>.
   * @alias module:model/GetTemplates402Response
   */
  function GetTemplates402Response() {
    _classCallCheck(this, GetTemplates402Response);

    GetTemplates402Response.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetTemplates402Response, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetTemplates402Response</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetTemplates402Response} obj Optional instance to populate.
     * @return {module:model/GetTemplates402Response} The populated <code>GetTemplates402Response</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetTemplates402Response();

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetTemplates402Response</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetTemplates402Response</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['message'] && !(typeof data['message'] === 'string' || data['message'] instanceof String)) {
        throw new Error("Expected the field `message` to be a primitive type in the JSON string but got " + data['message']);
      }

      return true;
    }
  }]);

  return GetTemplates402Response;
}();
/**
 * Error description
 * @member {module:model/GetTemplates402Response.MessageEnum} message
 */


GetTemplates402Response.prototype['message'] = undefined;
/**
 * Allowed values for the <code>message</code> property.
 * @enum {String}
 * @readonly
 */

GetTemplates402Response['MessageEnum'] = {
  /**
   * value: "Your account is suspended, please upgrade your account or contact support@pdfgeneratorapi.com"
   * @const
   */
  "Your account is suspended, please upgrade your account or contact support@pdfgeneratorapi.com": "Your account is suspended, please upgrade your account or contact support@pdfgeneratorapi.com"
};
var _default = GetTemplates402Response;
exports["default"] = _default;
