// pdf-generator-api-client/dist/model/InlineResponse402.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The InlineResponse402 model module.
 * @module model/InlineResponse402
 * @version 3.1.1
 */
var InlineResponse402 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InlineResponse402</code>.
   * @alias module:model/InlineResponse402
   */
  function InlineResponse402() {
    _classCallCheck(this, InlineResponse402);

    InlineResponse402.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(InlineResponse402, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>InlineResponse402</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InlineResponse402} obj Optional instance to populate.
     * @return {module:model/InlineResponse402} The populated <code>InlineResponse402</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InlineResponse402();

        if (data.hasOwnProperty('error')) {
          obj['error'] = _ApiClient["default"].convertToType(data['error'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return InlineResponse402;
}();
/**
 * Error description
 * @member {module:model/InlineResponse402.ErrorEnum} error
 */


InlineResponse402.prototype['error'] = undefined;
/**
 * HTTP Error code
 * @member {Number} status
 */

InlineResponse402.prototype['status'] = undefined;
/**
 * Allowed values for the <code>error</code> property.
 * @enum {String}
 * @readonly
 */

InlineResponse402['ErrorEnum'] = {
  /**
   * value: "Your account is suspended, please upgrade your account or contact support@pdfgeneratorapi.com"
   * @const
   */
  "Your account is suspended, please upgrade your account or contact support@pdfgeneratorapi.com": "Your account is suspended, please upgrade your account or contact support@pdfgeneratorapi.com"
};
var _default = InlineResponse402;
exports["default"] = _default;
