// pdf-generator-api-client/dist/model/MergeTemplate200ResponseMeta.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MergeTemplate200ResponseMeta model module.
 * @module model/MergeTemplate200ResponseMeta
 * @version 3.1.1
 */
var MergeTemplate200ResponseMeta = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MergeTemplate200ResponseMeta</code>.
   * @alias module:model/MergeTemplate200ResponseMeta
   */
  function MergeTemplate200ResponseMeta() {
    _classCallCheck(this, MergeTemplate200ResponseMeta);

    MergeTemplate200ResponseMeta.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MergeTemplate200ResponseMeta, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MergeTemplate200ResponseMeta</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MergeTemplate200ResponseMeta} obj Optional instance to populate.
     * @return {module:model/MergeTemplate200ResponseMeta} The populated <code>MergeTemplate200ResponseMeta</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MergeTemplate200ResponseMeta();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('display_name')) {
          obj['display_name'] = _ApiClient["default"].convertToType(data['display_name'], 'String');
        }

        if (data.hasOwnProperty('encoding')) {
          obj['encoding'] = _ApiClient["default"].convertToType(data['encoding'], 'String');
        }

        if (data.hasOwnProperty('content-type')) {
          obj['content-type'] = _ApiClient["default"].convertToType(data['content-type'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>MergeTemplate200ResponseMeta</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>MergeTemplate200ResponseMeta</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['display_name'] && !(typeof data['display_name'] === 'string' || data['display_name'] instanceof String)) {
        throw new Error("Expected the field `display_name` to be a primitive type in the JSON string but got " + data['display_name']);
      } // ensure the json data is a string


      if (data['encoding'] && !(typeof data['encoding'] === 'string' || data['encoding'] instanceof String)) {
        throw new Error("Expected the field `encoding` to be a primitive type in the JSON string but got " + data['encoding']);
      } // ensure the json data is a string


      if (data['content-type'] && !(typeof data['content-type'] === 'string' || data['content-type'] instanceof String)) {
        throw new Error("Expected the field `content-type` to be a primitive type in the JSON string but got " + data['content-type']);
      }

      return true;
    }
  }]);

  return MergeTemplate200ResponseMeta;
}();
/**
 * Document name. This value is automatically generated if name attribute is not defined in request.
 * @member {String} name
 */


MergeTemplate200ResponseMeta.prototype['name'] = undefined;
/**
 * Document name without the file extension.
 * @member {String} display_name
 */

MergeTemplate200ResponseMeta.prototype['display_name'] = undefined;
/**
 * Document encoding
 * @member {String} encoding
 */

MergeTemplate200ResponseMeta.prototype['encoding'] = undefined;
/**
 * Document content type.
 * @member {String} content-type
 */

MergeTemplate200ResponseMeta.prototype['content-type'] = undefined;
var _default = MergeTemplate200ResponseMeta;
exports["default"] = _default;
