// pdf-generator-api-client/dist/model/PaginationMeta.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PaginationMeta model module.
 * @module model/PaginationMeta
 * @version 4.0.3
 */
var PaginationMeta = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PaginationMeta</code>.
   * Information about pagination and navigation
   * @alias module:model/PaginationMeta
   */
  function PaginationMeta() {
    _classCallCheck(this, PaginationMeta);

    PaginationMeta.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PaginationMeta, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PaginationMeta</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PaginationMeta} obj Optional instance to populate.
     * @return {module:model/PaginationMeta} The populated <code>PaginationMeta</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PaginationMeta();

        if (data.hasOwnProperty('current_page')) {
          obj['current_page'] = _ApiClient["default"].convertToType(data['current_page'], 'Number');
        }

        if (data.hasOwnProperty('last_page')) {
          obj['last_page'] = _ApiClient["default"].convertToType(data['last_page'], 'Number');
        }

        if (data.hasOwnProperty('per_page')) {
          obj['per_page'] = _ApiClient["default"].convertToType(data['per_page'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>PaginationMeta</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PaginationMeta</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);

  return PaginationMeta;
}();
/**
 * @member {Number} current_page
 */


PaginationMeta.prototype['current_page'] = undefined;
/**
 * @member {Number} last_page
 */

PaginationMeta.prototype['last_page'] = undefined;
/**
 * @member {Number} per_page
 */

PaginationMeta.prototype['per_page'] = undefined;
var _default = PaginationMeta;
exports["default"] = _default;
