// pdf-generator-api-client/dist/model/TemplateParam.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _TemplateParamData = _interopRequireDefault(require("./TemplateParamData"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TemplateParam model module.
 * @module model/TemplateParam
 * @version 4.0.3
 */
var TemplateParam = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TemplateParam</code>.
   * Template id, version and data
   * @alias module:model/TemplateParam
   */
  function TemplateParam() {
    _classCallCheck(this, TemplateParam);

    TemplateParam.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TemplateParam, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TemplateParam</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TemplateParam} obj Optional instance to populate.
     * @return {module:model/TemplateParam} The populated <code>TemplateParam</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TemplateParam();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _TemplateParamData["default"].constructFromObject(data['data']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TemplateParam</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TemplateParam</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `data`
      if (data['data']) {
        // data not null
        _TemplateParamData["default"].validateJSON(data['data']);
      }

      return true;
    }
  }]);

  return TemplateParam;
}();
/**
 * Template ID
 * @member {Number} id
 */


TemplateParam.prototype['id'] = undefined;
/**
 * @member {module:model/TemplateParamData} data
 */

TemplateParam.prototype['data'] = undefined;
var _default = TemplateParam;
exports["default"] = _default;
